crates/store/src/state.rs

```diff
@@ -413,11 +449,71 @@ impl State {
             inner.blockchain.push(block_commitment);
         }
 
+        self.update_forest(account_updates, block_num).await?;
+
         info!(%block_commitment, block_num = block_num.as_u32(), COMPONENT, "apply_block successful");
 
         Ok(())
     }
 
+    /// Updates `SmtForest` with account deltas from a block
+    ///
+    /// This method updates the forest directly using the deltas extracted from the block.
+    ///
+    /// # Arguments
+    ///
+    /// * `account_updates` - Vector of (`AccountId`, `AccountUpdateDetails`) tuples from the block
+    /// * `block_num` - Block number for which these updates apply
+    ///
+    /// # Note
+    ///
+    /// - Private account updates are skipped as their state is not publicly visible.
+    /// - The number of changed accounts is implicitly bounded by the limited number of transactions
+    ///   per block.
+    #[instrument(target = COMPONENT, skip_all, fields(block_num = %block_num, num_accounts = account_updates.len()))]
+    async fn update_forest(
+        &self,
+        account_updates: Vec<(AccountId, AccountUpdateDetails)>,
+        block_num: BlockNumber,
+    ) -> Result<(), ApplyBlockError> {
```

Comment:
Related to the previous comment, we could pass just account deltas into this method. Also, we could probably make it a method on `InnerForest` - something like:
```rust
impl InnerForest {
    pub fn update(
        &mut self,
        updates: Vec<AccountDelta>,
        block_num: BlockNumber,
    ) -> Result<(), SomeError> {
        ...
    }
}
```